[ICE]: `expected wide pointer extra data (e.g. slice length or trait object vtable)` (missing lifetime)

<!--
[31mICE[0m: Rustc ./a.rs '' 'error: internal compiler error: compiler/rustc_const_eval/src/interpret/place.rs:36:17: expected wide pointer extra data (e.g. slice length or trait object vtable)', 'error: internal compiler error: compiler/rustc_const_eval/src/interpret/place.rs:36:17: expected wide pointer extra data (e.g. slice length or trait object vtable)'
File: /tmp/imm/a.rs
-->

auto-reduced (treereduce-rust):
````rust
struct Foo<'a>(<& /*'a*/ [fn()] as core::ops::Deref>::Target);
const _: *const Foo = 0 as _;

````

original:
````rust
struct Foo<'a>(<& /*'a*/ [fn()] as core::ops::Deref>::Target); // adding the lifetime solves the ice
const _: *const Foo = 0 as _;
fn main() {}

````

Version information
````
rustc 1.95.0-nightly (873b4beb0 2026-02-15)
binary: rustc
commit-hash: 873b4beb0cc726493b94c8ef21f68795c04fbbc1
commit-date: 2026-02-15
host: x86_64-unknown-linux-gnu
release: 1.95.0-nightly
LLVM version: 22.1.0

````

Possibly related line of code:
https://github.com/rust-lang/rust/blob/873b4beb0cc726493b94c8ef21f68795c04fbbc1/compiler/rustc_const_eval/src/interpret/place.rs#L30-L42

Command:
`/home/matthias/.rustup/toolchains/master/bin/rustc `

<details><summary><strong>Program output</strong></summary>
<p>

```
error[E0106]: missing lifetime specifier
 --> /tmp/icemaker_global_tempdir.XN1NnxVeArva/rustc_testrunner_tmpdir_reporting.1K5TD6vPH58s/mvce.rs:1:17
  |
1 | struct Foo<'a>(<& /*'a*/ [fn()] as core::ops::Deref>::Target);
  |                 ^ expected named lifetime parameter
  |
help: consider using the `'a` lifetime
  |
1 | struct Foo<'a>(<&'a  /*'a*/ [fn()] as core::ops::Deref>::Target);
  |                  ++

error[E0601]: `main` function not found in crate `mvce`
 --> /tmp/icemaker_global_tempdir.XN1NnxVeArva/rustc_testrunner_tmpdir_reporting.1K5TD6vPH58s/mvce.rs:2:30
  |
2 | const _: *const Foo = 0 as _;
  |                              ^ consider adding a `main` function to `/tmp/icemaker_global_tempdir.XN1NnxVeArva/rustc_testrunner_tmpdir_reporting.1K5TD6vPH58s/mvce.rs`

error: internal compiler error: /rustc-dev/873b4beb0cc726493b94c8ef21f68795c04fbbc1/compiler/rustc_const_eval/src/interpret/place.rs:36:17: expected wide pointer extra data (e.g. slice length or trait object vtable)


thread 'rustc' (626195) panicked at /rustc-dev/873b4beb0cc726493b94c8ef21f68795c04fbbc1/compiler/rustc_const_eval/src/interpret/place.rs:36:17:
Box<dyn Any>
stack backtrace:
   0:     0x7fbaf8bd449b - <<std[e84040df3bd3b365]::sys::backtrace::BacktraceLock>::print::DisplayBacktrace as core[6c6f0f72ed298c38]::fmt::Display>::fmt
   1:     0x7fbaf92110e0 - core[6c6f0f72ed298c38]::fmt::write
   2:     0x7fbaf8beb716 - <std[e84040df3bd3b365]::sys::stdio::unix::Stderr as std[e84040df3bd3b365]::io::Write>::write_fmt
   3:     0x7fbaf8baa548 - std[e84040df3bd3b365]::panicking::default_hook::{closure#0}
   4:     0x7fbaf8bc7903 - std[e84040df3bd3b365]::panicking::default_hook
   5:     0x7fbaf7b90f8f - std[e84040df3bd3b365]::panicking::update_hook::<alloc[d9717d1773e7bcdb]::boxed::Box<rustc_driver_impl[1d3373f79f63c38d]::install_ice_hook::{closure#1}>>::{closure#0}
   6:     0x7fbaf8bc7be2 - std[e84040df3bd3b365]::panicking::panic_with_hook
   7:     0x7fbaf7bc4ef1 - std[e84040df3bd3b365]::panicking::begin_panic::<rustc_errors[aa9f523260d7f0cb]::ExplicitBug>::{closure#0}
   8:     0x7fbaf7bbdda6 - std[e84040df3bd3b365]::sys::backtrace::__rust_end_short_backtrace::<std[e84040df3bd3b365]::panicking::begin_panic<rustc_errors[aa9f523260d7f0cb]::ExplicitBug>::{closure#0}, !>
   9:     0x7fbaf7bbdb0b - std[e84040df3bd3b365]::panicking::begin_panic::<rustc_errors[aa9f523260d7f0cb]::ExplicitBug>
  10:     0x7fbaf7bd0481 - <rustc_errors[aa9f523260d7f0cb]::diagnostic::BugAbort as rustc_errors[aa9f523260d7f0cb]::diagnostic::EmissionGuarantee>::emit_producing_guarantee
  11:     0x7fbaf819c439 - rustc_middle[11be9ef6dd95cb79]::util::bug::opt_span_bug_fmt::<rustc_span[8943e0009ed5ffe4]::span_encoding::Span>::{closure#0}
  12:     0x7fbaf819c5c2 - rustc_middle[11be9ef6dd95cb79]::ty::context::tls::with_opt::<rustc_middle[11be9ef6dd95cb79]::util::bug::opt_span_bug_fmt<rustc_span[8943e0009ed5ffe4]::span_encoding::Span>::{closure#0}, !>::{closure#0}
  13:     0x7fbaf818d41b - rustc_middle[11be9ef6dd95cb79]::ty::context::tls::with_context_opt::<rustc_middle[11be9ef6dd95cb79]::ty::context::tls::with_opt<rustc_middle[11be9ef6dd95cb79]::util::bug::opt_span_bug_fmt<rustc_span[8943e0009ed5ffe4]::span_encoding::Span>::{closure#0}, !>::{closure#0}, !>
  14:     0x7fbaf575fd94 - rustc_middle[11be9ef6dd95cb79]::util::bug::bug_fmt
  15:     0x7fbaf953f187 - <rustc_const_eval[30dfd147fcfea0c]::interpret::validity::ValidityVisitor<rustc_const_eval[30dfd147fcfea0c]::const_eval::machine::CompileTimeMachine>>::try_visit_primitive
  16:     0x7fbaf953f767 - <rustc_const_eval[30dfd147fcfea0c]::interpret::validity::ValidityVisitor<rustc_const_eval[30dfd147fcfea0c]::const_eval::machine::CompileTimeMachine> as rustc_const_eval[30dfd147fcfea0c]::interpret::visitor::ValueVisitor<rustc_const_eval[30dfd147fcfea0c]::const_eval::machine::CompileTimeMachine>>::visit_value
  17:     0x7fbaf953d319 - rustc_const_eval[30dfd147fcfea0c]::const_eval::eval_queries::intern_and_validate::<rustc_middle[11be9ef6dd95cb79]::mir::consts::ConstAlloc>
  18:     0x7fbafa287d42 - rustc_const_eval[30dfd147fcfea0c]::const_eval::eval_queries::eval_to_allocation_raw_provider
  19:     0x7fbafa285556 - rustc_query_impl[761ff2116f6b567]::query_impl::eval_to_allocation_raw::invoke_provider_fn::__rust_begin_short_backtrace
  20:     0x7fbafa284d21 - rustc_query_impl[761ff2116f6b567]::execution::try_execute_query::<rustc_middle[11be9ef6dd95cb79]::query::caches::DefaultCache<rustc_middle[11be9ef6dd95cb79]::ty::PseudoCanonicalInput<rustc_middle[11be9ef6dd95cb79]::mir::interpret::GlobalId>, rustc_middle[11be9ef6dd95cb79]::query::erase::ErasedData<[u8; 24usize]>>, {rustc_query_impl[761ff2116f6b567]::QueryFlags { is_anon: false, is_depth_limit: false, is_feedable: false }}, false>
  21:     0x7fbafa2848f3 - rustc_query_impl[761ff2116f6b567]::query_impl::eval_to_allocation_raw::get_query_non_incr::__rust_end_short_backtrace
  22:     0x7fbafa2808f2 - rustc_const_eval[30dfd147fcfea0c]::const_eval::eval_queries::eval_to_const_value_raw_provider
  23:     0x7fbafa280466 - rustc_query_impl[761ff2116f6b567]::query_impl::eval_to_const_value_raw::invoke_provider_fn::__rust_begin_short_backtrace
  24:     0x7fbafa27f3ff - rustc_query_impl[761ff2116f6b567]::execution::try_execute_query::<rustc_middle[11be9ef6dd95cb79]::query::caches::DefaultCache<rustc_middle[11be9ef6dd95cb79]::ty::PseudoCanonicalInput<rustc_middle[11be9ef6dd95cb79]::mir::interpret::GlobalId>, rustc_middle[11be9ef6dd95cb79]::query::erase::ErasedData<[u8; 24usize]>>, {rustc_query_impl[761ff2116f6b567]::QueryFlags { is_anon: false, is_depth_limit: true, is_feedable: false }}, false>
  25:     0x7fbafa27efb3 - rustc_query_impl[761ff2116f6b567]::query_impl::eval_to_const_value_raw::get_query_non_incr::__rust_end_short_backtrace
  26:     0x7fbaf9c3ea08 - <rustc_middle[11be9ef6dd95cb79]::ty::context::TyCtxt>::par_hir_body_owners::<rustc_hir_analysis[21cab81ddb5349c4]::check_crate::{closure#2}>::{closure#0}
  27:     0x7fbaf9c3e02e - rustc_hir_analysis[21cab81ddb5349c4]::check_crate
  28:     0x7fbaf9c41c96 - rustc_interface[ffa26b8aee337d6b]::passes::analysis
  29:     0x7fbafa469a19 - rustc_query_impl[761ff2116f6b567]::execution::try_execute_query::<rustc_middle[11be9ef6dd95cb79]::query::caches::SingleCache<rustc_middle[11be9ef6dd95cb79]::query::erase::ErasedData<[u8; 0usize]>>, {rustc_query_impl[761ff2116f6b567]::QueryFlags { is_anon: false, is_depth_limit: false, is_feedable: false }}, false>
  30:     0x7fbafa469754 - rustc_query_impl[761ff2116f6b567]::query_impl::analysis::get_query_non_incr::__rust_end_short_backtrace
  31:     0x7fbafa53a667 - <rustc_interface[ffa26b8aee337d6b]::passes::create_and_enter_global_ctxt<core[6c6f0f72ed298c38]::option::Option<rustc_interface[ffa26b8aee337d6b]::queries::Linker>, rustc_driver_impl[1d3373f79f63c38d]::run_compiler::{closure#0}::{closure#2}>::{closure#2} as core[6c6f0f72ed298c38]::ops::function::FnOnce<(&rustc_session[e077b3088d2cab4d]::session::Session, rustc_middle[11be9ef6dd95cb79]::ty::context::CurrentGcx, alloc[d9717d1773e7bcdb]::sync::Arc<rustc_data_structures[be2354717932db16]::jobserver::Proxy>, &std[e84040df3bd3b365]::sync::once_lock::OnceLock<rustc_middle[11be9ef6dd95cb79]::ty::context::GlobalCtxt>, &rustc_data_structures[be2354717932db16]::sync::worker_local::WorkerLocal<rustc_middle[11be9ef6dd95cb79]::arena::Arena>, &rustc_data_structures[be2354717932db16]::sync::worker_local::WorkerLocal<rustc_hir[e6f1d3e8b67d381c]::Arena>, rustc_driver_impl[1d3373f79f63c38d]::run_compiler::{closure#0}::{closure#2})>>::call_once::{shim:vtable#0}
  32:     0x7fbafa447ad0 - rustc_interface[ffa26b8aee337d6b]::interface::run_compiler::<(), rustc_driver_impl[1d3373f79f63c38d]::run_compiler::{closure#0}>::{closure#1}
  33:     0x7fbafa4307be - std[e84040df3bd3b365]::sys::backtrace::__rust_begin_short_backtrace::<rustc_interface[ffa26b8aee337d6b]::util::run_in_thread_with_globals<rustc_interface[ffa26b8aee337d6b]::util::run_in_thread_pool_with_globals<rustc_interface[ffa26b8aee337d6b]::interface::run_compiler<(), rustc_driver_impl[1d3373f79f63c38d]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>
  34:     0x7fbafa430ea0 - <std[e84040df3bd3b365]::thread::lifecycle::spawn_unchecked<rustc_interface[ffa26b8aee337d6b]::util::run_in_thread_with_globals<rustc_interface[ffa26b8aee337d6b]::util::run_in_thread_pool_with_globals<rustc_interface[ffa26b8aee337d6b]::interface::run_compiler<(), rustc_driver_impl[1d3373f79f63c38d]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>::{closure#1} as core[6c6f0f72ed298c38]::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
  35:     0x7fbafa431d2c - <std[e84040df3bd3b365]::sys::thread::unix::Thread>::new::thread_start
  36:     0x7fbafbce198b - <unknown>
  37:     0x7fbafbd65a0c - <unknown>
  38:                0x0 - <unknown>

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: please make sure that you have updated to the latest nightly

note: rustc 1.95.0-nightly (873b4beb0 2026-02-15) running on x86_64-unknown-linux-gnu

query stack during panic:
#0 [eval_to_allocation_raw] const-evaluating + checking `_`
#1 [eval_to_const_value_raw] simplifying constant for the type system `_`
#2 [analysis] running analysis passes on crate `mvce`
end of query stack
error: aborting due to 3 previous errors

Some errors have detailed explanations: E0106, E0601.
For more information about an error, try `rustc --explain E0106`.

```

</p>
</details>

<!--
query stack:
1 | struct Foo<'a>(<& /*'a*/ [fn()] as core::ops::Deref>::Target);
1 | struct Foo<'a>(<&'a  /*'a*/ [fn()] as core::ops::Deref>::Target);
#0 [eval_to_allocation_raw] const-evaluating + checking `_`
#1 [eval_to_const_value_raw] simplifying constant for the type system `_`
#2 [analysis] running analysis passes on crate `mvce`

-->

--- a/compiler/rustc_middle/src/ty/region.rs
+++ b/compiler/rustc_middle/src/ty/region.rs
@@ -291,7 +291,7 @@ impl<'tcx> Region<'tcx> {
             }
             ty::ReError(_) => {
                 flags = flags | TypeFlags::HAS_FREE_REGIONS;
-                flags = flags | TypeFlags::HAS_ERROR;
+                flags = flags | TypeFlags::HAS_RE_ERROR;
             }
         }
 

--- a/compiler/rustc_trait_selection/src/traits/project.rs
+++ b/compiler/rustc_trait_selection/src/traits/project.rs
@@ -659,7 +659,9 @@ fn project<'cx, 'tcx>(
         )));
     }
 
-    if let Err(guar) = obligation.predicate.error_reported() {
+    // We can still compute a projection type when there are only region errors,
+    // but type/const errors require early return.
+    if let Err(guar) = obligation.predicate.non_region_error_reported() {
         return Ok(Projected::Progress(Progress::error_for_term(
             selcx.tcx(),
             obligation.predicate,

--- a/compiler/rustc_type_ir/src/flags.rs
+++ b/compiler/rustc_type_ir/src/flags.rs
@@ -91,19 +91,24 @@ bitflags::bitflags! {
                                           | TypeFlags::HAS_TY_INHERENT.bits()
                                           | TypeFlags::HAS_CT_PROJECTION.bits();
 
+        /// Is a type or const error reachable?
+        const HAS_NON_REGION_ERROR          = 1 << 15;
+        /// Is a region error reachable?
+        const HAS_RE_ERROR                = 1 << 16;
         /// Is an error type/lifetime/const reachable?
-        const HAS_ERROR                   = 1 << 15;
+        const HAS_ERROR                   = TypeFlags::HAS_NON_REGION_ERROR.bits()
+                                          | TypeFlags::HAS_RE_ERROR.bits();
 
         /// Does this have any region that "appears free" in the type?
         /// Basically anything but `ReBound` and `ReErased`.
-        const HAS_FREE_REGIONS            = 1 << 16;
+        const HAS_FREE_REGIONS            = 1 << 17;
 
         /// Does this have any `ReBound` regions?
-        const HAS_RE_BOUND                = 1 << 17;
+        const HAS_RE_BOUND                = 1 << 18;
         /// Does this have any `Bound` types?
-        const HAS_TY_BOUND                = 1 << 18;
+        const HAS_TY_BOUND                = 1 << 19;
         /// Does this have any `ConstKind::Bound` consts?
-        const HAS_CT_BOUND                = 1 << 19;
+        const HAS_CT_BOUND                = 1 << 20;
         /// Does this have any bound variables?
         /// Used to check if a global bound is safe to evaluate.
         const HAS_BOUND_VARS              = TypeFlags::HAS_RE_BOUND.bits()
@@ -111,7 +116,7 @@ bitflags::bitflags! {
                                           | TypeFlags::HAS_CT_BOUND.bits();
 
         /// Does this have any `ReErased` regions?
-        const HAS_RE_ERASED               = 1 << 20;
+        const HAS_RE_ERASED               = 1 << 21;
 
         /// Does this value have parameters/placeholders/inference variables which could be
         /// replaced later, in a way that would change the results of `impl` specialization?
@@ -123,19 +128,19 @@ bitflags::bitflags! {
                                           | TypeFlags::HAS_CT_INFER.bits();
 
         /// Does this value have `InferTy::FreshTy/FreshIntTy/FreshFloatTy`?
-        const HAS_TY_FRESH                = 1 << 21;
+        const HAS_TY_FRESH                = 1 << 22;
 
         /// Does this value have `InferConst::Fresh`?
-        const HAS_CT_FRESH                = 1 << 22;
+        const HAS_CT_FRESH                = 1 << 23;
 
         /// Does this have any binders with bound vars (e.g. that need to be anonymized)?
-        const HAS_BINDER_VARS             = 1 << 23;
+        const HAS_BINDER_VARS             = 1 << 24;
 
         /// Does this type have any coroutines in it?
-        const HAS_TY_CORO                 = 1 << 24;
+        const HAS_TY_CORO                 = 1 << 25;
 
         /// Does this have have a `Bound(BoundVarIndexKind::Canonical, _)`?
-        const HAS_CANONICAL_BOUND         = 1 << 25;
+        const HAS_CANONICAL_BOUND         = 1 << 26;
     }
 }
 
@@ -240,7 +245,7 @@ impl<I: Interner> FlagComputation<I> {
             | ty::Str
             | ty::Foreign(..) => {}
 
-            ty::Error(_) => self.add_flags(TypeFlags::HAS_ERROR),
+            ty::Error(_) => self.add_flags(TypeFlags::HAS_NON_REGION_ERROR),
 
             ty::Param(_) => {
                 self.add_flags(TypeFlags::HAS_TY_PARAM);
@@ -489,7 +494,7 @@ impl<I: Interner> FlagComputation<I> {
                 }
             }
             ty::ConstKind::Expr(e) => self.add_args(e.args().as_slice()),
-            ty::ConstKind::Error(_) => self.add_flags(TypeFlags::HAS_ERROR),
+            ty::ConstKind::Error(_) => self.add_flags(TypeFlags::HAS_NON_REGION_ERROR),
         }
     }
 

--- a/compiler/rustc_type_ir/src/visit.rs
+++ b/compiler/rustc_type_ir/src/visit.rs
@@ -279,6 +279,8 @@ pub trait TypeVisitableExt<I: Interner>: TypeVisitable<I> {
 
     fn error_reported(&self) -> Result<(), I::ErrorGuaranteed>;
 
+    fn non_region_error_reported(&self) -> Result<(), I::ErrorGuaranteed>;
+
     fn has_non_region_param(&self) -> bool {
         self.has_type_flags(TypeFlags::HAS_PARAM - TypeFlags::HAS_RE_PARAM)
     }
@@ -352,6 +354,11 @@ pub trait TypeVisitableExt<I: Interner>: TypeVisitable<I> {
     fn still_further_specializable(&self) -> bool {
         self.has_type_flags(TypeFlags::STILL_FURTHER_SPECIALIZABLE)
     }
+
+    /// True if a type or const error is reachable
+    fn has_non_region_error(&self) -> bool {
+        self.has_type_flags(TypeFlags::HAS_NON_REGION_ERROR)
+    }
 }
 
 impl<I: Interner, T: TypeVisitable<I>> TypeVisitableExt<I> for T {
@@ -371,6 +378,18 @@ impl<I: Interner, T: TypeVisitable<I>> TypeVisitableExt<I> for T {
                 Err(guar)
             } else {
                 panic!("type flags said there was an error, but now there is not")
+            }
+        } else {
+            Ok(())
+        }
+    }
+
+    fn non_region_error_reported(&self) -> Result<(), I::ErrorGuaranteed> {
+        if self.has_non_region_error() {
+            if let ControlFlow::Break(guar) = self.visit_with(&mut HasErrorVisitor) {
+                Err(guar)
+            } else {
+                panic!("type flags said there was an non region error, but now there is not")
             }
         } else {
             Ok(())

--- a/tests/ui/dropck/dropck-after-failed-type-lowering.rs
+++ b/tests/ui/dropck/dropck-after-failed-type-lowering.rs
@@ -3,11 +3,12 @@
 trait B {
     type C<'a>;
     fn d<E>() -> F<E> {
+              //~^ ERROR: the trait bound `E: B` is not satisfied
         todo!()
     }
 }
 struct F<G> {
-    h: Option<<G as B>::C>,
+    h: Option<<G as B>::C>, //~ ERROR: the trait bound `G: B` is not satisfied
     //~^ ERROR missing generics for associated type `B::C`
 }
 

--- a/tests/ui/dropck/dropck-after-failed-type-lowering.stderr
+++ b/tests/ui/dropck/dropck-after-failed-type-lowering.stderr
@@ -1,5 +1,5 @@
 error[E0107]: missing generics for associated type `B::C`
-  --> $DIR/dropck-after-failed-type-lowering.rs:10:25
+  --> $DIR/dropck-after-failed-type-lowering.rs:11:25
    |
 LL |     h: Option<<G as B>::C>,
    |                         ^ expected 1 lifetime argument
@@ -14,6 +14,24 @@ help: add missing lifetime argument
 LL |     h: Option<<G as B>::C<'a>>,
    |                          ++++
 
-error: aborting due to 1 previous error
+error[E0277]: the trait bound `G: B` is not satisfied
+  --> $DIR/dropck-after-failed-type-lowering.rs:11:8
+   |
+LL |     h: Option<<G as B>::C>,
+   |        ^^^^^^^^^^^^^^^^^^^ the trait `B` is not implemented for `G`
+   |
+help: consider restricting type parameter `G` with trait `B`
+   |
+LL | struct F<G: B> {
+   |           +++
 
-For more information about this error, try `rustc --explain E0107`.
+error[E0277]: the trait bound `E: B` is not satisfied
+  --> $DIR/dropck-after-failed-type-lowering.rs:5:18
+   |
+LL |     fn d<E>() -> F<E> {
+   |                  ^^^^ the trait `B` is not implemented for `E`
+
+error: aborting due to 3 previous errors
+
+Some errors have detailed explanations: E0107, E0277.
+For more information about an error, try `rustc --explain E0107`.

--- a/tests/ui/unsized/thin-ptr-to-unsized-projection.rs
+++ b/tests/ui/unsized/thin-ptr-to-unsized-projection.rs
@@ -1,0 +1,6 @@
+// This is a regression test for <https://github.com/rust-lang/rust/issues/152682>
+struct Foo<'a>(<& /*'a*/ [fn()] as core::ops::Deref>::Target); // adding the lifetime solves the ice
+             //~^ ERROR: missing lifetime specifier [E0106]
+const _: *const Foo = 0 as _;
+                        //~^ ERROR: cannot cast `i32` to a pointer that is wide [E0606]
+fn main() {}

--- a/tests/ui/unsized/thin-ptr-to-unsized-projection.stderr
+++ b/tests/ui/unsized/thin-ptr-to-unsized-projection.stderr
@@ -1,0 +1,23 @@
+error[E0106]: missing lifetime specifier
+  --> $DIR/thin-ptr-to-unsized-projection.rs:2:17
+   |
+LL | struct Foo<'a>(<& /*'a*/ [fn()] as core::ops::Deref>::Target); // adding the lifetime solves the ice
+   |                 ^ expected named lifetime parameter
+   |
+help: consider using the `'a` lifetime
+   |
+LL | struct Foo<'a>(<&'a  /*'a*/ [fn()] as core::ops::Deref>::Target); // adding the lifetime solves the ice
+   |                  ++
+
+error[E0606]: cannot cast `i32` to a pointer that is wide
+  --> $DIR/thin-ptr-to-unsized-projection.rs:4:28
+   |
+LL | const _: *const Foo = 0 as _;
+   |                       -    ^ creating a `*const Foo<'_>` requires both an address and a length
+   |                       |
+   |                       consider casting this expression to `*const ()`, then using `core::ptr::from_raw_parts`
+
+error: aborting due to 2 previous errors
+
+Some errors have detailed explanations: E0106, E0606.
+For more information about an error, try `rustc --explain E0106`.

